### PR TITLE
Normalize drive letter case on windows

### PR DESCRIPTION
**Pull request by maxbrunsfeld**
Fixes https://github.com/atom/atom/issues/8551

### CODE REVIEW

**Comment by mnquintana** [comment]
Huh, that's super weird that `path.normalize` doesn't catch this. Is this something we should report to the Node team?

**Comment by 50Wliu** [comment]
I feel like drive letter capitalization shouldn't matter (see https://github.com/atom/atom/issues/3936#issuecomment-60172258), but if this fixes it, then yay?

File: src/browser/main.coffee [hunk 1]
```diff
@@ -54,17 +54,20 @@ start = ->
       else
         path.resolve(pathToOpen)
 
-    if args.devMode
-      AtomApplication = require path.join(args.resourcePath, 'src', 'browser', 'atom-application')
-    else
-      AtomApplication = require './atom-application'
+    AtomApplication = require path.join(args.resourcePath, 'src', 'browser', 'atom-application')
```
Comment (maxbrunsfeld) [inline src/browser/main.coffee]: @kevinsawicki Could you :eyes: this part? My goal removing this conditional was to cut off the problematic drive name at the earliest point possible, which is basically that `__filename` and `__dirname` and friends can be incorrect in `main.js`. This problem get propagated to other files if we call `require` with a relative path.

**Comment by maxbrunsfeld** [comment]
Ok this appears to fix the problem now.

**Pull request merged**